### ISSUE TITLE
Breakpoint hit -> window gets focused -> VS Code freezes for some time

### ISSUE THREAD

**Issue by MaxHausmann** [microsoft/vscode#123257]
- VS Code Version: 1.56.0
- OS Version: Windows_NT x64 10.0.18363

Steps to Reproduce:

1. Set breakpoint in js.-file
2. Launch debug-session in browser (all breakpoints in dev-mode disabled)
3. Switch to browser window
4. Maximize browser (because it is just half open)
5. Hit breakpoint
6. VSCode blinks in taskbar -> switch back to it
7. VSCode freezes and waits for something

**Hints:**
- By avoiding step 4, this issue doesn't occur.
- Minimizing and maximizing VSCode while its freezed solves this issue.
- This issue also happens in insider-version of VSCode.

Does this issue occur when all extensions are disabled?: Yes

Launch configuration (no, this problem is not related to brave, it happens also in google chrome):
```json
{
    "name": "Browser debug",
    "request": "launch",
    "type": "pwa-chrome",
    "runtimeExecutable": "C:/Program Files/BraveSoftware/Brave-Browser/Application/brave.exe",
    "runtimeArgs": ["--remote-debugging-port=9222"],
    "url": "http://127.0.0.1:5000",
    "webRoot": "${workspaceFolder}\\memory-viewer\\server",
    "pathMapping": {
        "/css": "${workspaceFolder}\\memory-viewer\\server\\static\\css",
        "/app": "${workspaceFolder}\\memory-viewer\\server\\static\\app",
        "/js": "${workspaceFolder}\\memory-viewer\\server\\static\\js"
    }       
},
```

**Comment by avenmore** [comment]
This looks like a regression of the issue mentioned in #116865, #117780, and others and declared a duplicate of #114459. It was fixed in 1.55 but it's back. Super annoying.

**Comment by MaxHausmann** [comment]
[vscode-debugadapter-b6c8efc1.json.gz](https://github.com/microsoft/vscode/files/6450011/vscode-debugadapter-b6c8efc1.json.gz)

**Comment by isidorn** [comment]
@MaxHausmann  can you reproduce if you start vscode with
`code --disable-features=CalculateNativeWinOcclusion <path to workspace/folder>`

**Comment by MaxHausmann** [comment]
With 
`code --disable-features=CalculateNativeWinOcclusion <path to workspace/folder>` this issue does not occur.
I tested it with several sizes of the browser-window (see initial text) and the breakpoint got hit instantly.

**Additonal information:** CMD says "Warning: 'disable-features' is not in the list of known options, but still passed to Electron/Chromium."

**Comment by isidorn** [comment]
@MaxHausmann lovely thanks a lot. 
This happens only on Windows, thus assigning `windows` flag, and it seems to be electron related thus assigning to my friend @deepak1556 

We will investigate if we can fix this as part of the recovery release. If not we plan to fix it in the future

**Comment by isidorn** [comment]
@MaxHausmann probably not related, but just to be 100% sure if you put
```json	
"debug.openDebug": "neverOpen"
```
In your settings, can you reproduce?



**WORKAROUND** 
Put in your settings
```json
"debug.focusWindowOnBreak": false
```

**Comment by MaxHausmann** [comment]
With 
```json "debug.openDebug": "neverOpen"``` I can reproduce the issue.

With 
```json
"debug.focusWindowOnBreak": false
```
As the (command says) VSCode window does not focus, but breakpoint got hit successfully.

**Comment by isidorn** [comment]
@MaxHausmann thanks, but if the windows does not get focus there is no freeze?
As for the `openDebug` setting that is something which we changed this milestone and I just wanted to verify it is not related to this issue.

**Comment by MaxHausmann** [comment]
Yes, there is no freeze. The icon in the taskbar is also not blinking.

**Comment by deepak1556** [comment]
Latest insiders has a workaround in place, you can launch vscode without the commandline flag `--disable-features=CalculateNativeWinOcclusion` to verify. I will keep this issue open however to investigate the root issue.

**Comment by isidorn** [comment]
Unassigning myself.
@deepak1556 Let me know if I can help somehow though

**Comment by kyrylkov** [comment]
This happens with Node.js also.  I suppose it's not going to be fixed in May 2021 release?  @deepak1556

**Comment by deepak1556** [comment]
This will be fixed in the May release, the issue is open to eliminate the workaround we have set.

**Comment by deepak1556** [comment]
The latest insiders https://code.visualstudio.com/insiders has updated runtime that does not contain any workaround for this issue. Can anyone check if the issue is present ? Thanks!

\closedWith 9520f0a515e7e24802c92f885a0efb242fca3831

**Comment by connor4312** [comment]
Per the linked issue this still seems to happen

**Comment by kombasltd** [comment]
Same here, was in vacation, did an update of VS Code when I returned and now I have this bug too. Never had this before.

**Comment by connor4312** [comment]
@deepak1556 Matt reported this on OSX

**Comment by deepak1556** [comment]
Yup that is definitely unrelated, as this occlusion tracker is specific to windows only. Tracking it here https://github.com/microsoft/vscode/issues/132142, currently waiting on logs to understand better.

**Comment by avenmore** [comment]
If I may note again as I did in a related thread, this problem is symptomatic of an unmatched [`LockWindowUpdate()`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-lockwindowupdate) pair - hopefully this might offer a clue.

**Comment by stefanoscerra** [comment]
I have the same issue with VS Code 1.60.1 and Windows 10 build 19043.

A working workaround is setting `"debug.focusWindowOnBreak": false`, as @isidorn mentioned.

**I also saw that if I grab the VS Code window by the title bar and then maximize it by dragging it to the top of the screen,
it unfreezes instantly.**

Please fix this, it's so annoying. Thank you.

**Comment by DerekSMorin** [comment]
I hit this all the time just opening any file.  I am not debugging.

**Comment by Tospaa** [comment]
Hello everyone,

I've confirmed the `--disable-features=CalculateNativeWinOcclusion` flag works well indeed, but is there a way to use this flag everytime I open a VS Code window?

I tried to add this line into the `~\.vscode\argv.json` file:
```json
"disable-features": "CalculateNativeWinOcclusion"
```
but it didn't work. Actually I did not expect this to work because it is clearly stated at the top of the file that 
> Only a subset of arguments is currently supported

I also tried to add `--disable-features=CalculateNativeWinOcclusion` flag into the `C:\Users\<user>\AppData\Local\Programs\Microsoft VS Code\bin\code.cmd` file as an argument. It works if I launch the VS Code from the terminal using `code .`, but it doesn't work when I launch the VS Code from the File Explorer context menu. I also added the same flag to these keys from the regedit:
- HKEY_CURRENT_USER\SOFTWARE\Classes\Directory\Background\shell\VSCode\command
- HKEY_CURRENT_USER\SOFTWARE\Classes\Directory\shell\VSCode\command

These changes make sure to have the CLI flag everytime I open VS Code, but I'm pretty sure those will be overridden with the updates. Even if they're not, I'd be happier if I can set the flag from one place to everywhere.

Is this possible somehow? Thanks in advance for any help!

**Comment by NathanBraslavski** [comment]
Any update? Thanks

**Comment by robert-niestroj** [comment]
Still have this on Windows 10 and latest VSCode version. Annoying :(

**Comment by kombasltd** [comment]
I can confirm that this behavior still exists.

**Comment by electrotype** [comment]
I am still having this issue too, on Windows 10 with the code running inside WSL2. I'm using the `"debug.focusWindowOnBreak": false` workaround, which is not perfect but ok.

If it can help, I noted that without `"debug.focusWindowOnBreak": false`, if you click on any open editor and then click on the editor containing the target breakpoint, then the breakpoint hits almost immediately... But that sucks because you need to know what editor to click, it doesn't show up automatically. Using the `Call Stack` is another trick to know where the hitted breakpoint is. But those are all workarounds looking for a real solution.

**Comment by mozhuanzuojing** [comment]
The latest version of Chrome browser has a similar phenomenon. I think this is Chromium's problem.

**Comment by lorand-horvath** [comment]
@deepak1556 I think the issues reported here don't have much to do with the problems I reported in #146737 because they only started with VSCode 1.66 and I haven't noticed any of that before.

**Comment by enunez** [comment]
This issue has been open for over a year now. Has there been any progress on fixing this bug? Is there an ETA? Has the root cause for the `CalculateNativeWinOcclusion` bug even been identified yet?

This issue still happens for me on Windows 11 with all the latest updates. It also happens when I simply leave the computer unattended for a while. VSCode hangs and takes ~30s to respond to input again. Same thing happens with certain pages opened in Chrome.

**Comment by skaldesh** [comment]
I am on Arch Linux with Sway WM:  
```
Version: 1.67.2
Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
Date: 2022-05-20T07:44:59.452Z
Electron: 17.4.3
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Linux x64 5.15.43-1-lts
```

I can confirm the bug there as well, that after switching focus from another application back to VSCode, it takes between 1-3s until VSCode becomes responsive again. I am happy to help debug, if need be.

**Comment by lorand-horvath** [comment]
Will investigate this further when Electron 18 is built into Insiders.

**Comment by lorand-horvath** [comment]
Solution: https://github.com/microsoft/vscode/issues/146737#issuecomment-1153128058

**Comment by hanct** [comment]
This issue still occur for v1.68.1

**Comment by lorand-horvath** [comment]
> This issue still occur for v1.68.1

@hanct Have you tried uninstalling completely and doing a clean install as I described here? https://github.com/microsoft/vscode/issues/146737#issuecomment-1153128058

**Comment by skaldesh** [comment]
@lorand-horvath   
This is no real solution, but a temporary fix.  
VSCode must be fixed for the real solution

**Comment by hanct** [comment]
> > This issue still occur for v1.68.1
> 
> @hanct Have you tried uninstalling completely and doing a clean install as I described here? https://github.com/microsoft/vscode/issues/146737#issuecomment-1153128058
> The problem is the cache growing to several thousands of files from the initial few hundred. So VSCode slows down to a crawl trying to process all of that.

I try to avoid that..cos when I start moving the code window (as suggested by others), the breakpoint appear. So I believe nothing to do with the cache sizes.

**Comment by hanct** [comment]
Then I also realize that sometime I don't even need to move the window, the break point response fast..but that is intermittent

**Comment by hanct** [comment]
For now, I try not to do anything so that when there is a fix, I have a broken vscode to prove this fix is working.

**Comment by Cr4zyEight** [comment]
There is a workaround on disabling this option in VS Code Settings.
By doing this u have to live with the fact that the VS Code icon will not blink in the taskbar when the debugger hits a breakpoint
But the Window will also not freeze
![image](https://user-images.githubusercontent.com/60257993/177943679-8aefb6bf-de8e-4e5b-9f5a-97bf16f90c85.png)
